sia/src/types/spendpolicy.rs

```diff
@@ -391,7 +391,9 @@ impl SiaDecodable for SpendPolicy {
 /// A policy that has been satisfied by a set of preimages and signatures.
 pub struct SatisfiedPolicy {
     pub policy: SpendPolicy,
+    #[serde(default)]
     pub preimages: Vec<Vec<u8>>,
+    #[serde(default)]
     pub signatures: Vec<Signature>,
```

Comment:
This change is not reflected in any tests. Probably worth adding a test case if you ran into an issue that caused you to add this.

Reply:
The gist is: if the Go type uses `omitempty` for the encoding, we need to decorate the Rust field with `serde(default)`, otherwise `serde` complains about missing fields. So basically every `Vec<>` we expect to JSON encode.